src/app/features/add-network/add-network.tsx

```diff
@@ -12,11 +12,12 @@ import { AddNetworkForm } from './add-network-form';
 import { useAddNetwork } from './use-add-network';
 
 export function AddNetwork() {
-  const { error, initialFormValues, loading, onSubmit } = useAddNetwork();
+  const { error, initialFormValues, loading, onSubmit, isEditNetworkMode } = useAddNetwork();
 
+  const title = isEditNetworkMode ? 'Edit network' : 'Add network';
   return (
     <>
-      <PageHeader title="Add network" />
+      <PageHeader title={title} />
```

Comment:
Maybe we can create a different `.layout` file for `add-network` and `edit-network`?

Or else pass a callback to set `title` to `<AddNetworkForm`? 

I'm not sure why we are using location state for `isEditNetworkMode`

Reply:
We can use a wrapper layout for the different pages and pass the form as `children`. 

We could also use a different Route URL if we wanted for `edit-network` 

We could set the title in `AddNetwork` by passing a callback to `<AddNetworkForm` then we don't need to use route state